instrumentation/ratpack/ratpack-1.7/library/src/test/groovy/io/opentelemetry/instrumentation/ratpack/client/InstrumentedHttpClientTest.groovy

```diff
@@ -232,4 +238,135 @@ class InstrumentedHttpClientTest extends Specification {
       attributes[HTTP_STATUS_CODE] == 200L
     }
   }
+
+  def "propagate http trace in ratpack services with compute thread"() {
+    expect:
+    def latch = new CountDownLatch(1)
+
+    def otherApp = EmbeddedApp.of { spec ->
+      spec.handlers {
+        it.get("foo") { ctx -> ctx.render("bar") }
+      }
+    }
+
+    def app = EmbeddedApp.of { spec ->
+      spec.registry(
+        Guice.registry { bindings ->
+          telemetry.configureServerRegistry(bindings)
+          bindings.bindInstance(HttpClient, telemetry.instrumentHttpClient(HttpClient.of(Action.noop())))
+          bindings.bindInstance(new BarService(latch, "${otherApp.address}foo", openTelemetry))
+        },
+      )
+      spec.handlers { chain ->
+        chain.get("foo") { ctx -> ctx.render("bar") }
+      }
+    }
+
+    app.address
+    latch.await()
+    def spanData = spanExporter.finishedSpanItems.find { it.name == "a-span" }
```

Comment:
I suspect that doing it this way will give you a flaky test that occasionally fails because spans haven't been received yet. Any reason why you are not using `assertTraces` as all other tests are?

Reply:
I am not familiar with the custom DSL and does not looks to be used in `groovy` tests
AFAIK, it have not been flakey in these tests

Reply:
This seems to be a weird test that doesn't extend `InstrumentationSpecification` so `assertTraces` isn't available. I guess you could add `new PollingConditions().eventually {}` around asserts as other tests in this file do.

Reply:
I've added the `eventually` as I am not too familiar with the `InMemorySpanExporter`
I think that those tests are deterministic with the `CountDownLatch`

Other tests in the file, are using eventually because the HttpClient is async and then needs `eventually`, but in this case, the `CountDownLatch` does the `countDown` in the Promise subscription which makes it deterministic. 

The only scenario would be if the publishing to `InMemorySpanExporter` is async. As I am not too familiar with the InMemory, let's leave the `eventually`?